### ISSUE TITLE
Update vg image to version v1.4.0-1744-gea8edfd

### ISSUE THREAD

**Issue by cmarkello** [BD2KGenomics/cgl-docker-lib#207]

**Comment by cmarkello** [comment]
@cket @fnothaft @hannes-ucsc The built vg docker image is still not uploaded to the quay.io even after the jenkins build completes. How long does it take for the CI to complete?

**Comment by hannes-ucsc** [comment]
http://jenkins.cgcloud.info/job/cgl-docker-lib/323/consoleFull

```
DEBUG:__main__:vg/['make', 'push']: g++ -O3 -msse4.1 -fopenmp -std=c++11 -ggdb -g -c -o obj/sha1.o deps/sha1/sha1.cpp -I/home/vg/include -I. -I/home/vg/src -I/home/vg/src/unittest -I/home/vg/src/subcommand -I/home/vg/cpp -I/home/vg/include/dynamic -I/home/vg/include/sonLib && cp deps/sha1/*.h* /home/vg/include/
DEBUG:__main__:vg/['make', 'push']: . ./source_me.sh && g++ -O3 -msse4.1 -fopenmp -std=c++11 -ggdb -g -c -o obj/main.o src/main.cpp -I/home/vg/include -I. -I/home/vg/src -I/home/vg/src/unittest -I/home/vg/src/subcommand -I/home/vg/cpp -I/home/vg/include/dynamic -I/home/vg/include/sonLib -ggdb -L/home/vg/lib -lvcflib -lgssw -lssw -lprotobuf -lhts -ljansson -lncurses -lgcsa2 -lxg -ldivsufsort -ldivsufsort64 -lvcfh -lgfakluge -lraptor2 -lsupbub -lsdsl -lpinchesandcacti -l3edgeconnected -lsonlib -ltcmalloc_minimal -Wl,-rpath,/home/vg/lib   -lrocksdb  -lpthread -lrt -lsnappy -lz -lbz2 
DEBUG:__main__:vg/['make', 'push']: . ./source_me.sh && g++ -O3 -msse4.1 -fopenmp -std=c++11 -ggdb -g -c -o obj/gssw_aligner.o src/gssw_aligner.cpp -I/home/vg/include -I. -I/home/vg/src -I/home/vg/src/unittest -I/home/vg/src/subcommand -I/home/vg/cpp -I/home/vg/include/dynamic -I/home/vg/include/sonLib -ggdb -L/home/vg/lib -lvcflib -lgssw -lssw -lprotobuf -lhts -ljansson -lncurses -lgcsa2 -lxg -ldivsufsort -ldivsufsort64 -lvcfh -lgfakluge -lraptor2 -lsupbub -lsdsl -lpinchesandcacti -l3edgeconnected -lsonlib -ltcmalloc_minimal -Wl,-rpath,/home/vg/lib   -lrocksdb  -lpthread -lrt -lsnappy -lz -lbz2 
DEBUG:__main__:vg/['make', 'push']: [91msrc/gssw_aligner.cpp: In member function 'void vg::Aligner::align_internal(vg::Alignment&, std::vector<vg::Alignment>*, vg::Graph&, int64_t, bool, int32_t, bool)':
DEBUG:__main__:vg/['make', 'push']: src/gssw_aligner.cpp:167:85: error: too few arguments to function 'gssw_graph_mapping** gssw_graph_trace_back_pinned_multi(gssw_graph*, gssw_node*, int32_t, const char*, int32_t, int8_t*, int8_t*, uint8_t, uint8_t, int8_t)'
DEBUG:__main__:vg/['make', 'push']:                                                                         gap_extension);
DEBUG:__main__:vg/['make', 'push']:                                                                                      ^
DEBUG:__main__:vg/['make', 'push']: [0m[91mIn file included from src/gssw_aligner.hpp:10:0,
DEBUG:__main__:vg/['make', 'push']:                  from src/gssw_aligner.cpp:1:
DEBUG:__main__:vg/['make', 'push']: /home/vg/include/gssw.h:516:22: note: declared here
DEBUG:__main__:vg/['make', 'push']:  gssw_graph_mapping** gssw_graph_trace_back_pinned_multi (gssw_graph* graph,
DEBUG:__main__:vg/['make', 'push']:                       ^
DEBUG:__main__:vg/['make', 'push']: [0m[91msrc/gssw_aligner.cpp: In member function 'void vg::QualAdjAligner::align_pinned(vg::Alignment&, vg::Graph&, int64_t, bool)':
DEBUG:__main__:vg/['make', 'push']: src/gssw_aligner.cpp:896:89: error: too few arguments to function 'gssw_graph_mapping* gssw_graph_trace_back_pinned_qual_adj(gssw_graph*, gssw_node*, const char*, const char*, int32_t, int8_t*, int8_t*, uint8_t, uint8_t, int8_t)'
DEBUG:__main__:vg/['make', 'push']:                                                                      scaled_gap_extension);
DEBUG:__main__:vg/['make', 'push']:                                                                                          ^
DEBUG:__main__:vg/['make', 'push']: In file included from src/gssw_aligner.hpp:10:0,
DEBUG:__main__:vg/['make', 'push']:                  from src/gssw_aligner.cpp:1:
DEBUG:__main__:vg/['make', 'push']: /home/vg/include/gssw.h:503:21: note: declared here
DEBUG:__main__:vg/['make', 'push']:  gssw_graph_mapping* gssw_graph_trace_back_pinned_qual_adj (gssw_graph* graph,
DEBUG:__main__:vg/['make', 'push']:                      ^
DEBUG:__main__:vg/['make', 'push']: [0mMakefile:229: recipe for target 'obj/gssw_aligner.o' failed
DEBUG:__main__:vg/['make', 'push']: [91mmake: *** [obj/gssw_aligner.o] Error 1
DEBUG:__main__:vg/['make', 'push']: [0mThe command '/bin/sh -c . ./source_me.sh && make static' returned a non-zero code: 2
DEBUG:__main__:vg/['make', 'push']: make: *** [runtime/vg] Error 1
DEBUG:__main__:vg/['make', 'push']: 
INFO:__main__:Running "['make', 'push']" for tool quinine.
INFO:__main__:Running "['make', 'push']" for tool quinine succeeded!
INFO:__main__:Running "['make', 'push']" for tool rnaseq-cgl-pipeline.
ERROR:__main__:Running "['make', 'push']" for tool rnaseq-cgl-pipeline FAILED with code 2! Output:
```

What I don't understand is why the build succeeds even though ``make push`` clearly fails for one of the projects. The error propagation in this project is still a mess. It's not up to me anymore to assign work here. Please file a ticket and ping @jvivian, @cket or @fnothaft for resolution.

Maybe @natanlao wants to get his fingers dirty?

**Comment by hannes-ucsc** [comment]
s/a mess/broken/

**Comment by fnothaft** [comment]
Yeah, that's a bug. That being said, the relevant line of code (https://github.com/BD2KGenomics/cgl-docker-lib/blame/master/jenkins.py#L189-L190) was unchanged in #200, which cleaned up most of the logging and tool discovery code in `jenkins.py`. From a quick glance at the code, I think the bug is that `if cmd == 'make':` should be `if cmd == ['make']:` or something similar. `cmd` is a list, so `if cmd == 'make':` will always evaluate to false.

@natanlao let me know if you'd like to take this on, else I'll take more of a look.

**Comment by cmarkello** [comment]
I was able to build it locally. I'll test the build locally again. I don't believe there should have been a build failure.

**Comment by natanlao** [comment]
@fnothaft Sorry, I didn't see this!

**Comment by fnothaft** [comment]
@natanlao no problem, it was a pretty quick fix!

**Comment by fnothaft** [comment]
Closed by #213.
